gax-grpc/src/main/java/com/google/api/gax/grpc/CallContext.java

```diff
@@ -67,11 +71,41 @@ public CallOptions getCallOptions() {
     return callOptions;
   }
 
-  public CallContext withChannel(Channel channel) {
-    return new CallContext(channel, this.callOptions);
+  public GrpcCallContext withChannel(Channel channel) {
+    return new GrpcCallContext(channel, this.callOptions);
   }
 
-  public CallContext withCallOptions(CallOptions callOptions) {
-    return new CallContext(this.channel, callOptions);
+  public GrpcCallContext withCallOptions(CallOptions callOptions) {
+    return new GrpcCallContext(this.channel, callOptions);
+  }
+
+  @Override
+  public <RequestT, ResponseT> UnaryCallable<RequestT, ResponseT> newUnaryCallable(
```

Comment:
It turns out this won't be necessary at all - I created it an an earlier state of refactoring, but now the correct objects can be fully instantiated within core, so it doesn't need to defer to the transport-specific implementation any more. I will remove it.